relay-server/src/services/project_cache.rs

```diff
@@ -1268,4 +1309,89 @@ mod tests {
             assert!(buffer_rx.try_recv().is_err())
         }
     }
+
+    #[tokio::test]
+    async fn periodic_unspool() {
```

Comment:
@jjbayer  Added a test, if you want to have a look.

Reply:
Nice!